### PR TITLE
docs: roadmap 2024 H1 blog

### LINKED ISSUE #7939: docs/feat: a public roadmap for the Ibis project
### Please describe the issue

several users have requested a public roadmap for the Ibis project. this helps plan their adoption (e.g. awareness of a large SQLglot refactor or big new features they may be waiting to adopt) and instills confidence in the longevity of the project

we currently have a GitHub project with a roadmap view we should consider making public in a few months after we are confident in our use of it 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
## Description of changes

de-drafts the roadmap blog and makes some edits

## Issues closed

extra closes #7939

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
recommend we merge this on 2/15 in the morning (pending review/approval, to some extent done in #8184)

File: docs/posts/roadmap-2024-H1/index.qmd
Comment (cpcloud) [inline docs/posts/roadmap-2024-H1/index.qmd:140]:
```suggestion
early) state. We will further invest in IbisML this year to get it to a
```

File: docs/posts/roadmap-2024-H1/index.qmd
Comment (cpcloud) [inline docs/posts/roadmap-2024-H1/index.qmd:200]:
```suggestion
We have a few other themes for 2024 that are lower priority.
```

File: docs/posts/roadmap-2024-H1/index.qmd
Comment (cpcloud) [inline docs/posts/roadmap-2024-H1/index.qmd:247]:
```suggestion
This writeup of our roadmap is heavily focused on Q1 of 2024. Looking beyond,
```

Biased feels like the wrong word to me here.

**Review by cpcloud — CHANGES_REQUESTED** [review]

**Review by cpcloud — APPROVED** [review]

**Comment by cpcloud** [comment]
Thanks!

**Pull request merged**
